src/pi_cam.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file
+ * @brief     Implementation for Pi camera
+ * @author    Willem de Groot, Arsalan Anwari
+ * @license   See LICENSE
+ */
+#include "pi_cam.hpp"
+
+Camera::PiCam::PiCam(const char *folderPath, const Vector2D resolution, int fps)
+    : Camera(fps, resolution), folderPath(folderPath), buffer(resolution.getX() * resolution.getY() * 3) {
+    correctPath();
+}
+
+void Camera::PiCam::correctPath() {
+    if (folderPath.back() != '/') {
+        folderPath += '/';
+    }
+}
+
+void Camera::PiCam::clearCommandBuffer(std::stringstream &parser) {
+    parser.str(std::string());
+    parser.clear();
+}
+
+std::string Camera::PiCam::getSettings() {
+    std::stringstream result;
+    result << " -fps " << fps << " -w " << resolution.getX() << " -h " << resolution.getY();
+    return result.str();
+}
+
+void Camera::PiCam::printSettings() {
+    std::cout << "Resolution: " << getResolution().print() << "\n"

```

Comment:
std::endl is preferred over \n as it flushes, unless there is a performance reason not to use it.